Sources/Puree/Output.swift

```diff
@@ -2,13 +2,22 @@ import Foundation
 
 public typealias OutputOptions = [String: Any]
 
-public struct OutputSetting {
+public protocol OutputSettingProtocol {
+    func makeOutput(_ logStore: LogStore) throws -> Output
```

Comment:
Similar to `FilterSettingProtocol`, wouldn't the following make things simpler and clearer?
```swift
public protocol OutputSettingProtocol {
    var outputMaker: (_ logStore: LogStore) throws -> Output { get }
}
```

Reply:
In fact your `OutputSetting` could be just a `(_ logStore: LogStore) -> Output`...